Adds Plasma Stunner

SiegDerMaus opened the pull request:
Adds a weapon intended to bridge the gap between tasers and lethals. For
now, it will be adminspawn only, and will undergo some testing. It can
be mapped in later.

SiegDerMaus committed:
Adds Plasma Stunner

Adds a weapon intended to bridge the gap between tasers and lethals. For
now, it will be adminspawn only, and will undergo some testing. It can
be mapped in later.

SiegDerMaus commented:
My apologies for the two previous pull requests about this, they were done incorrectly thanks to my Git Desktop client messing up, pointing to the wrong stuff, etc.

comma commented:
So... taser before it was made non-instadrop?
With some minor burn damage and not working in space, but still.

SiegDerMaus commented:
Yeah, it's going to need tweaking so it's not OP, but I can't do that on my own. It also has an incredibly limited range. When firing it, it makes it to about the halfway point between your character and the edge of the screen. It also has fewer shots, and a significantly longer firing delay.

MistakeNot4892 commented on code/modules/projectiles/projectile/energy.dm:
```diff
@@ -117,3 +117,51 @@
+					ear_safety += 1
+				if(istype(M:head, /obj/item/clothing/head/helmet))
+					ear_safety += 1
+		if(ear_safety > 0)
```
The second condition will never evaluate since if >1 is true, >0 is true by definition.

MistakeNot4892 commented on code/modules/projectiles/projectile/energy.dm:
```diff
@@ -117,3 +117,51 @@
+			M.Stun(5)
+			M.Weaken(2)
+		else if (ear_safety > 1)
+			M.Stun(2)
```
You should probably check armour or some such to make it possible to resist this, otherwise it's just a retread of the old taser problem.

SiegDerMaus replied:
There is a protection check. If the target is wearing a helmet or is a hulk, it's reduced by half, if they have both and/or are wearing earmuffs, it's reduced to almost insignificant.

PsiOmegaDelta commented on code/modules/projectiles/projectile/energy.dm:
```diff
@@ -117,3 +117,51 @@
+		var/ear_safety = 0
+		if(iscarbon(M))
+			if(ishuman(M))
+				if(istype(M:l_ear, /obj/item/clothing/ears/earmuffs) || istype(M:r_ear, /obj/item/clothing/ears/earmuffs))
```
Colons are evil. Add ```var/mob/living/carbon/human/H = M``` above and use H.l_ear, etc. instead.

scrdest commented on code/modules/projectiles/projectile/energy.dm:
```diff
@@ -117,3 +117,51 @@
+	damage_type = BURN
+	vacuum_traversal = 0
+
+	proc/bang(var/mob/living/carbon/M)
```
Relative path.

Raptor1628 commented:
Does it have a spread? Or does it only have one projectile? Because I could see this like a crowd control weapon, like the PEPS from Deus Ex.

SiegDerMaus commented:
It only targets one person.

SiegDerMaus committed:
Changes and Tweaks

Reduces armour penetration, adds a missing part of if statement, makes
requested changes, clarifies ambiguous comment.

SiegDerMaus commented:
Another thing, if colons are really that bad, that proc is literally just a pared down version of flashbang code, so maybe it should be changed there as well.

SiegDerMaus committed:
Fixes a silly logic error

Don't judge me, I was tired. >.>

scrdest commented:
Why, yes it should. It just sometimes *isn't yet*.

PsiOmegaDelta commented on code/modules/projectiles/projectile/energy.dm:
```diff
@@ -117,3 +117,52 @@
+	damage_type = BURN
+	vacuum_traversal = 0
+
+	proc/bang(var/mob/living/carbon/M)
```
It is indeed relative pathing, please correct this.

SiegDerMaus committed:
Fixes Relative Pathing

This will teach me to adapt existing code and expect it to be up to
coding standards.
